lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -154,10 +166,49 @@ def apply_config(config)
       @excluded_patterns.concat(config["excluded_patterns"]) if config["excluded_patterns"]
       @included_patterns.concat(config["included_patterns"]) if config["included_patterns"]
       @excluded_magic_comments.concat(config["excluded_magic_comments"]) if config["excluded_magic_comments"]
+      @fnmatch_exclude_pattern = compute_fnmatch_exclude_pattern
     end
 
     private
 
+    sig { returns(String) }
+    def compute_fnmatch_exclude_pattern
+      relative_patterns = @excluded_patterns
+        .select { |p| p.end_with?("/**/*") }
+        .map { |p| p.delete_prefix("#{Dir.pwd}/") }
```

Comment:
That makes sense, though I don't think we want the `!` on delete_prefix. We're allocating a new array, but the strings are still the ones in `@excluded_patterns`, which means we'd be mutating the content of `@excluded_patterns`.

Reply:
I added that change to the `select` line, but kept what I had on the `map` line, since I don't think we want to mutate `@excluded_patterns`' strings. Let me know if you think otherwise.
https://github.com/Shopify/ruby-lsp/pull/2082/commits/3a8c7fe6c2e42373f9a2508fa3dddfc36fd66d80

Reply:
Just to clarify, `included_patterns` should be relative to `Dir.pwd`, right? `apply_config` doesn't automatically prefix it with `Dir.pwd`, and I don't see how users would realistically set `Dir.pwd` in the YAML, considering they'd all have the repo checked out to different folders. Feels like we need to be prepending `Dir.pwd` to every entry in `included_patterns`. It also feels like `excluded_patterns` should similarly be expected to be within `Dir.pwd`. I'm struggling to understand what files someone would be including / excluding that aren't in a repo or a gem. But from what I can tell, excluded_patterns refers solely to application code, not gems. You exclude entire gems, not paths within a gem.

Right now I'm feeling like `include_patterns` doesn't even work, unless the user provides an absolute path to the folder on disk, which isn't very portable. What's the use-case for this? Understanding will help me figure out how to best optimize this, because the goal is to avoid calling `Dir.glob` on folders that are excluded, which means we need to take the included patterns, apply the exclude patterns that end with wildcards against them, so cut down on which folders `Dir.glob` scans.

Reply:
That's good point. Now that I think of it, all patterns have to be relative to something, since you can clone a project in any directory.

I'll bring this up with the team, but I think we should indeed make everything relative to `Dir.pwd`.

Reply:
If we're including `File.join(Dir.pwd, "**", "*.rb")` by default, all I can really think that people might be including is non-`.rb` ruby files, like a Gemfile or Guardfile. So that leaves me really wondering what people even use `include_patterns` for. When people specify `include_patterns`, it doesn't seem to remove the default pattern. So anything they're including is likely already in the existing pattern, unless it doesn't end with `.rb` or is outside of Dir.pwd.